src/hud.cpp

```diff
@@ -849,10 +872,451 @@ namespace openre::hud
+        const auto inventorySize = gGameTable.inventory_size;
+        const auto mixQuantity = redItem.Quantity + greenItem.Quantity;
+
+        gGameTable.byte_691F76 = 0;
+        switch (gGameTable.byte_691F64)
+        {
+        case INVENTORY_MIX_ITEM_MOVE_GREEN_CURSOR:
+        {
+            auto oldGreenCursor = greenCursor;
+
+            if (gGameTable.fg_system & 0x80000000)
+            {
+                // Right
+                if (gGameTable.word_9885FC & 0x2000 && greenCursor < inventorySize - 1)
+                {
+                    if (greenItem.Part == 1)
+                    {
+                        greenCursor++;
+                    }
+                    greenCursor++;
+                }
+                // Left
+                if (gGameTable.word_9885FC & 0x8000 && greenCursor && greenCursor != INVENTORY_SPECIAL_ITEM_SLOT)
+                {
+                    auto auxGreenCursor = greenCursor;
+                    if (greenItem.Part == 2)
+                    {
+                        if (greenCursor == 1)
+                        {
+                            auxGreenCursor = 2;
+                        }
+                        else
+                        {
+                            auxGreenCursor--;
+                        }
+                    }
+                    greenCursor = --auxGreenCursor;
+                }
+                // Down
+                if (gGameTable.word_9885FC & 0x4000)
+                {
+                    if (greenCursor >= inventorySize - 2)
+                    {
+                        if (greenCursor == INVENTORY_SPECIAL_ITEM_SLOT)
+                        {
+                            greenCursor = 1;
+                        }
+                    }
+                    else
+                    {
+                        greenCursor += 2;
+                    }
+                }
+                // Up
+                if (gGameTable.word_9885FC & 0x1000 && greenCursor != INVENTORY_SPECIAL_ITEM_SLOT)
+                {
+                    if (greenCursor <= 1)
+                    {
+                        greenCursor = INVENTORY_SPECIAL_ITEM_SLOT;
+                    }
+                    else
+                    {
+                        greenCursor -= 2;
+                    }
+                }
+                if (oldGreenCursor != greenCursor)
+                {
+                    snd_se_on(0x4040000);
+                }
+            }
+            // Mix items
+            if (gGameTable.key_trg & 0x1000)
+            {
+                gGameTable.byte_691F64 = hud_check_item_mix();
+                if (gGameTable.byte_691F64)
+                {
+                    snd_se_on(0x4060000);
+                }
+                else
+                {
+                    snd_se_on(0x4070000);
+                }
+            }
+            // Cancel mix
+            if (gGameTable.key_trg & 0x2000)
```

Comment:
Should use your new input enum. Might also be worth having a function just called e.g. `check_input(BUTTON_CANCEL)`